src/main/java/fr/insalyon/creatis/gasw/executor/batch/internals/BatchFile.java

```diff
@@ -0,0 +1,44 @@
+package fr.insalyon.creatis.gasw.executor.batch.internals;
+
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class BatchFile {
+
+    final private BatchJobData  data;
+    final private StringBuilder builder = new StringBuilder(1024);
+
+    public StringBuilder build() {
+        switch (data.getConfig().getOptions().getBatchEngine()) {
+            case PBS:
+                doPBS();
+                break;
+            case SLURM:
+                doSlurm();
+                break;
+        }
+        doCommon();
+
+        return builder;
```

Comment:
return directly the String, as I don't thing some string should still be appended afterwards.